service/history/queues/executor_wrapper.go

```diff
@@ -24,74 +24,9 @@
 
 package queues
 
-import (
-	"context"
-
-	"go.temporal.io/server/common/headers"
-	"go.temporal.io/server/common/log"
-	"go.temporal.io/server/common/log/tag"
-	"go.temporal.io/server/common/metrics"
-	"go.temporal.io/server/common/namespace"
-)
-
 type (
-	executorWrapper struct {
-		currentClusterName string
-		registry           namespace.Registry
-		activeExecutor     Executor
-		standbyExecutor    Executor
-		logger             log.Logger
+	ExecutorWrapper interface {
```

Comment:
nit: I think Decorator is more precise here because decorators are always f(T) -> T, but an adapter is also a wrapper, and so are many other things